packages/governance/src/multiCandidateVoteCounter.js

```diff
@@ -0,0 +1,249 @@
+import { makePromiseKit } from '@endo/promise-kit';
+import {
+  buildQuestion,
+  ChoiceMethod,
+  coerceQuestionSpec,
+  ElectionType,
+  positionIncluded,
+} from './question.js';
+import { scheduleClose } from './closingRule.js';
+import {
+  VoteCounterAdminI,
+  VoteCounterCloseI,
+  VoteCounterPublicI,
+} from './typeGuards.js';
+import { makeQuorumCounter } from './quorumCounter.js';
+import { breakTie } from './breakTie.js';
+
+const { details: X } = assert;
+
+const validateQuestionSpec = questionSpec => {
+  coerceQuestionSpec(questionSpec);
+
+  questionSpec.electionType === ElectionType.ELECTION ||
+    questionSpec.electionType === ElectionType.SURVEY ||
+    assert.fail(X`${questionSpec.electionType} must be ELECTION or SURVEY`);
+
+  questionSpec.method === ChoiceMethod.PLURALITY ||
+    assert.fail(X`${questionSpec.method} must be PLURALITY`);
+};
+
+/** @type {BuildVoteCounter} */
+const makeMultiCandidateVoteCounter = (
+  questionSpec,
+  threshold,
+  instance,
+  publisher,
+) => {
+  validateQuestionSpec(questionSpec);
+
+  const question = buildQuestion(questionSpec, instance);
+  const details = question.getDetails();
+
+  let isOpen = true;
+  const positions = questionSpec.positions;
+  const maxChoices = questionSpec.maxChoices;
+
+  /** @type { PromiseRecord<Position[]> } */
+  const outcomePromise = makePromiseKit();
+  /** @type { PromiseRecord<VoteStatistics> } */
+  const tallyPromise = makePromiseKit();
+
+  /**
+   * @typedef {object} RecordedBallot
+   * @property {Position[]} chosen
+   * @property {bigint} shares
+   */
+  /** @type {Store<Handle<'Voter'>,RecordedBallot> } */
+  const allBallots = makeStore('voterHandle');
+
+  const countVotes = () => {
+    assert(!isOpen, X`can't count votes while the election is open`);
+
+    let spoiled = 0n;
+    const tally = Array(positions.length).fill(0n);
+
+    for (const { chosen, shares } of allBallots.values()) {
+      for (const position of chosen) {
+        const positionIndex = positions.findIndex(p => keyEQ(p, position));
+        if (positionIndex < 0) {
+          spoiled += shares;
+        } else {
+          tally[positionIndex] += shares;
+        }
+      }
+    }
+
+    /** @type { VoteStatistics } */
+    const stats = {
+      spoiled,
+      votes: allBallots.getSize(),
+      results: tally.map((_, i) => ({
+        position: positions[i],
+        total: tally[i],
+      })),
+    };
+
+    tallyPromise.resolve(stats);
+
+    if (!makeQuorumCounter(threshold).check(stats)) {
+      outcomePromise.reject('No quorum');
+      /** @type {OutcomeRecord} */
+      const voteOutcome = {
+        question: details.questionHandle,
+        outcome: 'fail',
+        reason: 'No quorum',
+      };
+      E(publisher).publish(voteOutcome);
+      return;
+    }
+
+    const sortedPositions = stats.results.sort((a, b) => {
+      if (a.total < b.total) return 1;
+      if (a.total > b.total) return -1;
+      return 0;
+    });
```

Comment:
Does that do anything different from this?

```suggestion
    const sortedPositions = stats.results.sort((a, b) => b.total - a.total);
```

Reply:
it returns `TypeError: can't convert BigInt value to Number value` so the subtraction won't work

Reply:
```suggestion
    const sortedPositions = stats.results.sort((a, b) => Number(b.total - a.total));
```